pkg/virtualfs/dir_entry.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package virtualfs
+
+import (
+	"os"
+	"time"
+
+	"github.com/kopia/kopia/fs"
+)
+
+// dirEntry is mock in-memory implementation of a directory dirEntry
+type dirEntry struct {
+	name    string
+	mode    os.FileMode
+	size    int64
+	modTime time.Time
+	owner   fs.OwnerInfo
+}
```

Comment:
Please put type assertion vars for the Kopia interfaces dirEntry is supposed to satisfy. In this case I think it is fs.Entry:
```
var _ fs.Entry = (*dirEntry)(nil)
```

Reply:
These are in the last file below.